File: 1603_Design_Parking_System/1603.py
class ParkingSystem:

    def __init__(self, big: int, medium: int, small: int):
        self.size = [big, medium, small]

    def addCar(self, carType: int) -> bool:
        if self.size[carType - 1] > 0:
            self.size[carType - 1] -= 1
            return True
        return False


# Your ParkingSystem object will be instantiated and called as such:
# obj = ParkingSystem(big, medium, small)
# param_1 = obj.addCar(carType)

obj = ParkingSystem(1, 1, 0)
for i in [1, 2, 3, 1]:
    print(obj.addCar(i))

File: 1603_Design_Parking_System/1603.cpp
#include <bits/stdc++.h>
using namespace std;

class ParkingSystem {
private:
    int num_b = 0, num_m = 0, num_s = 0;
    const int BIG = 1, MEDIUM = 2, SMALL = 3;
public:
    ParkingSystem(int big, int medium, int small) {
        num_b = big;
        num_m = medium;
        num_s = small;
    }
    
    bool addCar(int carType) {
        if (carType == BIG) {
            if (num_b > 0) {
                num_b--;
                return true;
            }
            return false;
        }
        else if (carType == MEDIUM) {
            if (num_m > 0) {
                num_m--;
                return true;
            }
            return false;
        }
        else {
            if (num_s > 0) {
                num_s--;
                return true;
            }
            return false;
        }
    }
};

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * ParkingSystem* obj = new ParkingSystem(big, medium, small);
 * bool param_1 = obj->addCar(carType);
 */

int main() {
    ParkingSystem* obj = new ParkingSystem(1, 1, 0);
    vector<int> car = {1, 2, 3, 1};
    for (const int c: car) {
        cout << obj->addCar(c) << " ";
    }
    cout << "\n";
    return 0;
}